hal/src/common/mod.rs

```diff
@@ -9,6 +9,10 @@ pub mod time;
 pub mod timer_params;
 pub mod timer_traits;
 
+#[cfg(feature = "unproven")]
```

Comment:
Maybe an `#[cfg(all(feature = "dma", feature = "unproven"))]` instead of two cfg directives?